utils/src/main/kotlin/storage/HttpFileStorage.kt

```diff
@@ -0,0 +1,102 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.utils.storage
+
+import com.here.ort.utils.OkHttpClientHelper
+import com.here.ort.utils.getUserOrtDirectory
+
+import java.io.IOException
+import java.io.InputStream
+import java.net.HttpURLConnection
+import java.util.concurrent.TimeUnit
+
+import okhttp3.CacheControl
+import okhttp3.Headers.Companion.toHeaders
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+
+/**
+ * A [FileStorage] that stores files on an HTTP server.
+ */
+class HttpFileStorage(
+    /**
+     * The URL to store files at.
+     */
+    private val url: String,
+
+    /**
+     * Custom headers that are added to all HTTP requests.
+     */
+    private val headers: Map<String, String> = emptyMap(),
+
+    /**
+     * A path inside the [ORT user directory][getUserOrtDirectory] used for caching HTTP responses. Defaults to
+     * "cache/http".
+     */
+    private val cachePath: String = "cache/http",
+
+    /**
+     * The max age of an HTTP cache entry in seconds. Defaults to 0 which always validates the cached response with the
+     * remote server.
+     */
+    private val cacheMaxAgeInSeconds: Int = 0
+) : FileStorage {
+    override fun read(path: String): InputStream {
+        val request = Request.Builder()
+            .headers(headers.toHeaders())
+            .cacheControl(CacheControl.Builder().maxAge(cacheMaxAgeInSeconds, TimeUnit.SECONDS).build())
+            .get()
+            .url(pathUrl(path))
+            .build()
+
+        val response = OkHttpClientHelper.execute(cachePath, request)
+        if (response.code == HttpURLConnection.HTTP_OK) {
+            response.body?.let { body ->
+                return body.byteStream()
+            }
+
+            response.close()
+            throw IOException("The response body must not be null.")
+        }
+
+        response.close()
+        throw IOException("Could not read from '${pathUrl(path)}': ${response.code} - ${response.message}")
+    }
+
+    override fun write(path: String, inputStream: InputStream) {
+        inputStream.use {
+            val request = Request.Builder()
+                .headers(headers.toHeaders())
+                .put(it.readAllBytes().toRequestBody())
+                .url(pathUrl(path))
+                .build()
+
+            return OkHttpClientHelper.execute(cachePath, request).use { response ->
+                if (response.code != HttpURLConnection.HTTP_CREATED) {
+                    throw IOException(
+                        "Could not store file at '${pathUrl(path)}': ${response.code} - ${response.message}"
+                    )
+                }
+            }
+        }
+    }
+
+    private fun pathUrl(path: String) = "$url/$path"
```

Comment:
When I first read the function name at the call site I couldn't tell from its name what it's doing. And I wondered what complex path transformation would justify a function here, where in reality the string template is hardly longer than the function name. So, bottom line, I see no value in this being a function, but on the contrary, for me it's harder to understand what the calling code does (because I need to look at one more code location).

Reply:
But in the end, this should not block the merge, so I've approved anyway.

Reply:
To me the benefit of using the function is that the logic to build the URL for a path is in a single place instead of duplicating it in the read and write functions. This makes it easier to change the logic, e.g. if validation of the path needs to be added.